client/gutenberg/editor/controller.js

```diff
@@ -42,6 +48,36 @@ function registerDataPlugins( userId ) {
 	use( plugins.controls );
 }
 
+export const jetpackBlocki18n = ( context, next ) => {
+	if ( ! isEnabled( 'gutenberg/block/jetpack-preset' ) ) {
+		return next();
+	}
+
+	const state = context.store.getState();
+	const localeSlug = getCurrentLocaleSlug( state );
+
+	// We don't need to localize English
+	if ( localeSlug === 'en' ) {
+		return next();
+	}
+
+	const languageFileUrl =
+		'https://widgets.wp.com/languages/jetpack-gutenberg-blocks/' + localeSlug + '.json';
+
+	request.get( languageFileUrl ).end( ( error, response ) => {
```

Comment:
Minor terminology nit, this isn't a synchronous call (as laid out in the title and description of the PR), as it doesn't block the browser UI while it's inflight. That said, we should not block the browser UI by using a true synchronous request, so this is perfectly fine, unless you're expecting it to block the browser...

Reply:
Yeah, good point, this is really sync-ish, because we don't prevent browser from loading anything, but in the same time we won't call `next()` until the translations request is finished.